File: Example/第三方框架/第三方框架/Classes/Other/Category/UIBarButtonItem+Extension.m
//
//  UIBarButtonItem+Extension.m
//  第三方框架
//
//  Created by 金亮齐 on 2017/3/17.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

#import "UIBarButtonItem+Extension.h"

@implementation UIBarButtonItem (Extension)

+ (instancetype)itemWithImage:(NSString *)image highImage:(NSString *)highImage target:(id)target action:(SEL)action {
    //设置按钮
    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
    [button setBackgroundImage:[UIImage imageNamed:image] forState:UIControlStateNormal];
    [button setBackgroundImage:[UIImage imageNamed:highImage] forState:UIControlStateHighlighted];
    [button addTarget:target action:action forControlEvents:UIControlEventTouchUpInside];
    button.size = button.currentBackgroundImage.size;
    return [[self alloc] initWithCustomView:button];
}

@end

File: 项目/Swift项目/百思不得姐/百思不得姐/Classes/Other(其他)/Category(扩展)/UIBarButtonItem+Extension.swift
//
//  UIBarButtonItem+Extension.swift
//  百思不得姐
//
//  Created by 金亮齐 on 2017/11/10.
//  Copyright © 2017年 金亮齐. All rights reserved.
//

import UIKit

extension UIBarButtonItem {
    
    convenience init(image: String, highImage: String, targat:Any, action: Selector)  {
        let button = UIButton(type: .custom)
        button.setBackgroundImage(UIImage(named:image), for: .normal)
        button.setBackgroundImage(UIImage(named:highImage), for: .highlighted)
        button.addTarget(targat, action: action, for: .touchUpInside)
        button.sizeToFit()
        self.init(customView: button)
    }
    
}

